### PR TITLE
Revert "Use config-reload container as an init container"

### PR DESCRIPTION
**Pull request by timja**
Reverts jenkinsci/helm-charts#926

### CODE REVIEW

File: charts/jenkins/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v2
 name: jenkins
 home: https://jenkins.io/
-version: 4.7.0
+version: 4.6.7
```
Comment (timja) [inline charts/jenkins/Chart.yaml:4]:
```suggestion
version: 4.7.1
```

**Comment by dduportal** [comment]
For info: https://github.com/jenkins-infra/kubernetes-management/pull/4528#issue-1937908539

**Comment by dominykas** [comment]
> The init container was stuck trying to emit requests to the JCasc endpoint

Can you confirm some details here? It's not entirely clear to me from the logs here. Do you have a custom config for number of retries, maybe?

I will fix the calls to the endpoint, of course, because that's a bug for sure, but I'm trying to understand what was actually happening here. We're using defaults for retries, etc, so it just exited after the default 5 retries, but it was successful and booted up OK.

The `METHOD` env var duplication is also something I can fix, but it's also not something that should prevent the thing from starting?

Thanks!

**Comment by dduportal** [comment]
> > The init container was stuck trying to emit requests to the JCasc endpoint
> 
> Can you confirm some details here? It's not entirely clear to me from the logs here. Do you have a custom config for number of retries, maybe?

Hi @dominykas , you can find the values we are using in this helmfile value file: https://github.com/jenkins-infra/kubernetes-management/blob/main/config/jenkins_infra.ci.jenkins.io.yaml . We also have secrets values merged with this but it is only the `controller.additionalSecrets` key.

There was no retry setup as far as I can tell (but I might forget something).


What happened is that we deployed the 4.7.0 version of the jenkins chart in https://github.com/jenkins-infra/kubernetes-management/pull/4516/files and the controller never went back once applied.

As explained in my comment above, the pod was stuck in the init phase with the container "config-init-reload" running. Checking its logs, we saw a bunch of what I put above: errors when trying to connect to the JCasc endpoint at localhost:8080.

After 35m, we gave up and had to rollback

=> does it clarify the issue we had?


> I will fix the calls to the endpoint, of course, because that's a bug for sure, but I'm trying to understand what was actually happening here. We're using defaults for retries, etc, so it just exited after the default 5 retries, but it was successful and booted up OK.


Something which is not clear to me is : why does the config-init-reload init container tries to connect to the JCasc endpoint? It makes sense to have the init container preparing the whole configuration set by placing files on the proper location, etc.
But since it is an init container, the JCasc endpoint provided in the `jenkins` container cannot exist during the whole init container phase.
What did I miss?

**Comment by dduportal** [comment]
> I think I understand this now. This must have had to do with your `METHOD=SLEEP`, as you correctly pointed out - there's a warning about the duplicate value. I guess the last value takes precedence, so instead of exiting, the init container went to sleep.
> 
> I'm about to submit the fix, although not sure if there's an easy way we can test it before merging?
> 
> > Something which is not clear to me is : why does the config-init-reload init container tries to connect to the JCasc endpoint?
> 
> My oversight. In my local testing I had that disabled, but when I submitted the final PR, I missed that. #931 will fix that.

oh good, it make sense and maps to what we saw!

Not sure how to easily reproduce it on a local k3s but we don't mind testing the chart with your change before the release though on our production if it helps ? (It is a private controller of the official Jenkins infrastructure). We can do it on a call

**Comment by dominykas** [comment]
It would be great, if you have the option, to try out https://github.com/jenkinsci/helm-charts/pull/931. I still want to make some improvements to the code/output, but it won't change the semantics.

**Pull request merged**